geo-types/src/coordinate.rs

```diff
@@ -25,11 +27,37 @@ use approx::{AbsDiffEq, RelativeEq, UlpsEq};
 /// [vector space]: //en.wikipedia.org/wiki/Vector_space
 #[derive(Eq, PartialEq, Clone, Copy, Debug, Hash, Default)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct Coordinate<T: CoordNum> {
+pub struct Coordinate<T: CoordNum, Z: ZCoord = NoValue, M: Measure = NoValue> {
     pub x: T,
     pub y: T,
+    pub z: Z,
+    pub m: M,
+}
+
+impl<T: CoordNum, Z: ZCoord, M: Measure> Coordinate<T, Z, M> {
+    pub fn new(x: T, y: T, z: Z, m: M) -> Self {
```

Comment:
What do you think about adding docs here encouraging the user to utilize the `coord!` macro?